media-libs/loadpng/loadpng-1.5.recipe

```diff
@@ -1,33 +1,45 @@
-DESCRIPTION="
-loadpng is some glue that makes it easy to use libpng to load and save bitmaps \
-from Allegro programs.
-"
+SUMMARY="Reference library for working with PNG files"
+DESCRIPTION="Some glue that makes it easy to use libpng to load and save bitmaps from \
+Allegro programs."
 HOMEPAGE="http://tjaden.strangesoft.net/loadpng/"
-SOURCE_URI="http://tjaden.strangesoft.net/loadpng/loadpng-1.5.tar.gz"
-REVISION="1"
-STATUS_HAIKU="stable"
-DEPEND="media-libs/allegro >= 4.4
-		media-libs/libpng >=1.2"
-MESSAGE="This might not build with libpng >= 1.4"
-CHECKSUM_MD5="5e9ecdf1762c3fca47477bf0263b96ba"
+COPYRIGHT="1997-2004 Psyk Software."
+LICENSE="Public Domain"
+SOURCE_URI="http://tjaden.strangesoft.net/loadpng/loadpng-$portVersion.tar.gz"
+REVISION="2"
+CHECKSUM_SHA256="ba0e496e5368de70e6df43cb5500468cda402e63ab95b6a466e29ec6b795b7d7"
+SOURCE_DIR="loadpng-$portVersion"
+
+ARCHITECTURES="x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+    loadpng$secondaryArchSuffix = $portVersion
+    "
+REQUIRES="
+    haiku$secondaryArchSuffix
+    lib:libpng16$secondaryArchSuffix
+    lib:allegro$secondaryArchSuffix
+    "
+
+BUILD_REQUIRES="
+    haiku${secondaryArchSuffix}_devel
+    cmd:gcc$secondaryArchSuffix
+    cmd:make
+    "
+
 BUILD()
 {
-	cd loadpng-1.5
-	make
+    make
 }
 
 INSTALL()
 {
-	cd loadpng-1.5
-	if [ -n ${DESTDIR} ];then
-		mkdir -p ${DESTDIR}/boot/common/lib
-		mkdir -p ${DESTDIR}/boot/common/include
-		make install DESTDIR=${DESTDIR}
-	else
-		make install
-	fi
-	make install
+    if [ -n ${DESTDIR} ];then
+        mkdir -p ${DESTDIR}/boot/common/lib
+        mkdir -p ${DESTDIR}/boot/common/include
+        make install DESTDIR=${DESTDIR}
+    else
+        make install
+    fi
+    make install
```

Comment:
You should be able to drop the `if` statements and just use `make install DESTDIR=$prefix` here